Visitor can see collection of recipes

elvitak opened the pull request:
### This PR contains:

- add index serializer
- add index route
- add recipe controller with index action
- add recipe model
- add date formatter to serializer
- add method to sort recipes by newly created

https://www.pivotaltracker.com/story/show/181136258

@Mljungho @DefyCab @giacoletti

elvitak committed:
add basic spec

elvitak committed:
add basic spec

DefyCab committed:
add namespace and index route

DefyCab committed:
generate recipe model with title and instructions

elvitak committed:
add recipe controller with index action, add model spec and factory, validates presence of title and instructions

DefyCab committed:
add non-lazy evaluation to factory create recipe

elvitak committed:
add index serializer, add association spec in model spec, add ctive_model_serializers gem

elvitak committed:
change association to validations

giacoletti commented on spec/serializers/recipe/index_serializer_spec.rb:
```diff
@@ -0,0 +1,16 @@
+
+  it 'is expected to wrap data in a key reflecting the resource name' do
+    expect(subject.keys).to match ['recipes']
+  end
```
```  
  it 'is expected to include relevant keys' do
    expected_keys = %w[id title instructions]
    expect(subject['recipes'].first.keys).to match expected_keys
  end

  it 'is expected to contain keys with values of specific data types' do
    expect(subject['recipes'].first).to match(
      {
        'id' => an_instance_of(Integer),
        'title' => an_instance_of(String),
        'instructions' => an_instance_of(Text),
        'created_at' => an_instance_of(...),
        'updated_at' => an_instance_of(...),
      }
    )
  end
  ```

giacoletti commented on app/controllers/api/recipes_controller.rb:
```diff
@@ -0,0 +1,6 @@
+class Api::RecipesController < ApplicationController
+  def index
+    recipes = Recipe.all
```
Did we agree that we are gonna return the latest recipes?
Also a limit should be in place, like the last 30-20 recipes

DefyCab replied:
Yes we did! I thought it might be out of scope, but let's do it now.

giacoletti commented on app/serializers/recipe/index_serializer.rb:
```diff
@@ -0,0 +1,3 @@
+class Recipe::IndexSerializer < ActiveModel::Serializer
+  attributes :id, :title, :instructions, :created_at, :updated_at
```
On the client we would like to receive a humanized date format.
You can either choose a new name like:
```
  def published
    object.created_at.to_formatted_s(:long)
  end
  ```
  or use the same "created_at" name:
  ```
  def created_at
    object.created_at.to_formatted_s(:long)
  end
  ```

DefyCab committed:
update index controller to sort by latest created recipe
add specs to index serializer
add date formatter to index serializer
add scope to recipe model to enable sorting by date

DefyCab committed:
Update index_serializer_spec.rb
